homeassistant/components/ultrasync/sensor.py

```diff
@@ -0,0 +1,176 @@
+"""Monitor the Interlogix/Hills ComNav UltraSync Hub."""
+
+import logging
+from typing import Callable, List
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import UltraSyncEntity
+from .const import (
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DOMAIN,
+    SENSOR_UPDATE_LISTENER,
+    SENSORS,
+)
+from .coordinator import UltraSyncDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up UltraSync sensor based on a config entry."""
+    coordinator: UltraSyncDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    # At least one sensor must be pre-created or Home Assistant will not
+    # call any updates
+    hass.data[DOMAIN][entry.entry_id][SENSORS]["area01_state"] = UltraSyncSensor(
+        coordinator, entry.entry_id, entry.data[CONF_NAME], "area01_state", "Area1State"
+    )
+
+    async_add_entities([hass.data[DOMAIN][entry.entry_id][SENSORS]["area01_state"]])
+
+    @callback
+    def _auto_manage_sensors(areas: dict, zones: dict) -> None:
+        """Dynamically create/delete sensors based on what was detected by the hub."""
+
+        _LOGGER.debug(
+            "Entering _auto_manage_sensors with %d area(s), and %d zone(s)",
+            len(areas),
+            len(zones),
+        )
+
+        # our list of sensors to add
+        new_sensors = []
+
+        # A pointer to our sensors
+        sensors = hass.data[DOMAIN][entry.entry_id][SENSORS]
+
+        # Track our detected sensors (for automatic updates if required)
+        detected_sensors = set()
+
+        for meta in areas:
+            bank_no = meta["bank"]
+            sensor_id = "area{:0>2}_state".format(bank_no + 1)
+            detected_sensors.add(sensor_id)
+            if sensor_id not in sensors:
+                # hash our entry
+                sensors[sensor_id] = UltraSyncSensor(
```

Comment:
Don't store the entity. If we need to ask the entity to do something from outside the entity we use our dispatch helper.